test/tests/cat_tests.py

```diff
@@ -26,10 +26,22 @@ def test_cat(self):
         cat_cmd.errfile = io.StringIO()
 
         arg_parser = arguments.get_parser()
-        arg_sets = (['cat', str(test.id), 'build.sh'],)
+        arg_sets = (['cat', str(test.id), 'run.tmpl'],)
+        true_out="""
+#!/bin/bash
+
+# The first (and only) argument of the build script is the test id.
+export TEST_ID={}
+export PAV_CONFIG_FILE={}
+source {}
+
+# Perform the sequence of test commands.
+echo "Hello World."
+""".format(test.id, test.TODO, test.TODO)
 
         for arg_set in arg_sets:
             args = arg_parser.parse_args(arg_set)
             cat_cmd.run(self.pav_cfg, args)
-            output.fprint(cat_cmd.outfile)
-            output.fprint(cat_cmd.errfile)
+            # FIXME: always true
+            if cat_cmd.outfile != true_out:
+                return errno.ENOMSG
```

Comment:
Test always passes with this line.

Reply:
cat_cmd.outfile is a file-like object. You have to read it. `cat_cmd.outfile.read()`

Reply:
Use the test object assert statements. 
```python
self.assertEqual(thing1, thing2)
#In this case:
true_out = (test.path/arg_set[-1]).open().read()
self.assertEqual(cat_cmd.outfile.read(), true_out)
```